movement/napari/loader_widgets.py

```diff
@@ -177,19 +226,54 @@ def _on_load_clicked(self):
 
     def _format_data_for_layers(self):
         """Extract data required for the creation of the napari layers."""
-        # Load data as a movement dataset
-        loader = (
-            load_poses
-            if self.source_software in SUPPORTED_POSES_FILES
-            else load_bboxes
-        )
-        ds = loader.from_file(self.file_path, self.source_software, self.fps)
+        if self.source_software in SUPPORTED_NETCDF_FILES:
+            # Add logic for netCDF files
+            try:
+                ds = xr.open_dataset(self.file_path)
+            except Exception as e:
+                show_error(f"Error opening netCDF file: {e}")
+                return False
+
+            # Get the dataset type from its attributes
+            ds_type = ds.attrs.get("ds_type", None)
+            validator = {
+                "poses": ValidPosesDataset,
+                "bboxes": ValidBboxesDataset,
+            }.get(ds_type)
+
+            if validator:
+                try:
+                    # Using validate_dataset function (currently Private)
```

Comment:
Again, sorry for being pedantic, but this comment should be removed (it just repeats the code line that follows).

```suggestion
                    # Using validate_dataset function (currently Private)
```